src/uct/ib/dc/base/dc_ep.h

```diff
@@ -166,7 +166,11 @@ static inline void uct_dc_iface_dci_put_dcs(uct_dc_iface_t *iface, uint8_t dci)
      * move the group to the 'wait for dci alloc' state
      */
     ucs_arbiter_group_desched(uct_dc_iface_tx_waitq(iface), &ep->arb_group);
-    ucs_arbiter_group_schedule(uct_dc_iface_dci_waitq(iface), &ep->arb_group);
+    if (ep->fc.fc_wnd > 0) {
```

Comment:
nice catch.
does it make sense to extract this code to common function?
is it possible to reproduce this bug with a unit test?